src/python/grpcio/grpc/experimental/aio/_channel.py

```diff
@@ -13,15 +13,209 @@
 # limitations under the License.
 """Invocation-side implementation of gRPC Asyncio Python."""
 import asyncio
-from typing import Callable, Optional
+from typing import Callable, Dict, Optional
 
+import grpc
 from grpc import _common
 from grpc._cython import cygrpc
 
 SerializingFunction = Callable[[str], bytes]
 DeserializingFunction = Callable[[bytes], str]
 
 
+class AioRpcError(grpc.RpcError):
+    """An RpcError to be used by the asynchronous API."""
+
+    def __init__(self,
+                 code: grpc.StatusCode,
+                 details: Optional[str] = None,
+                 initial_metadata: Optional[Dict] = None,
+                 trailing_metadata: Optional[Dict] = None):
+        super().__init__(self)
+        self._code = code
+        self._details = details
+        self._initial_metadata = initial_metadata
+        self._trailing_metadata = trailing_metadata
+
+    def code(self) -> grpc.StatusCode:
+        """
+        Returns:
+          The `grpc.StatusCode` status code.
+        """
+        return self._code
+
+    def details(self) -> Optional[str]:
+        """
+        Returns:
+          The description of the error.
+        """
+        return self._details
+
+    def initial_metadata(self) -> Optional[Dict]:
+        """
+        Returns:
+          The inital metadata received.
+        """
+        return self._initial_metadata
+
+    def trailing_metadata(self) -> Optional[Dict]:
+        """
+        Returns:
+          The trailing metadata received.
+        """
+        return self._trailing_metadata
+
+
+class Call:
+    """ Object for managing RPC calls, returned when an instance of
+    `UnaryUnaryMultiCallable` object is called.
+    """
+
+    _cancellation_details = 'Locally cancelled by application!'
+
+    def __init__(self, aio_call: asyncio.Task,
+                 response_deserializer: DeserializingFunction,
+                 aio_call_cancel_status: cygrpc.AioCancelStatus) -> None:
+        self._exception = None
+        self._response = None
+        self._code = None
+        self._details = None
+        self._initial_metadata = None
+        self._trailing_metadata = None
+        self._cancelled = False
+        self._aio_call = aio_call
```

Comment:
Nit: Seems a bit redundant to have `aio` in the name. It's already in a file and class specifically for asyncio.